lib/index.js

```diff
@@ -190,23 +188,16 @@ class ElectronDownloader {
     })
   }
 
-  createTempDir (cb, onSuccess) {
-    this.tmpdir = path.join(os.tmpdir(), `electron-tmp-download-${process.pid}-${Date.now()}`)
-    fs.mkdirs(this.tmpdir, (err) => {
-      if (err) return cb(err)
-      onSuccess(cb)
-    })
-  }
-
   downloadChecksum (cb) {
-    this.downloadFile(this.checksumUrl, this.checksumFilename, this.cachedChecksum, cb, this.verifyChecksum.bind(this))
+    this.downloadFile(this.checksumUrl, this.cachedChecksum, cb, this.verifyChecksum.bind(this))
   }
 
-  downloadFile (url, filename, cacheFilename, cb, onSuccess) {
-    debug('downloading', url, 'to', this.tmpdir)
+  downloadFile (url, cacheFilename, cb, onSuccess) {
+    const tempFileName = `tmp-${process.pid}-${(tmpFileCounter++).toString(16)}-${path.basename(cacheFilename)}`
```

Comment:
I'd rather have this use something like [tempnam](https://www.npmjs.com/package/tempnam) instead of introducing global variables and hand-rolling it.